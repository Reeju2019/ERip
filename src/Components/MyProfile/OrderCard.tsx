import * as React from 'react'
import './MyProfile.css'
import MyProfileData from '../../Data/MyProfile.mock.json'

interface IOrderCardProps {
  item: {
    order_id: string
    time: string
    device: string
    issue: string
    order_status: string
    image: string
  }
}

const OrderCard: React.FunctionComponent<IOrderCardProps> = ({ item }) => {
  return (
    <>
      <section className='bg-white p-3 orderCard'>
        <div className='d-flex justify-content-between mt-3'>
          <div className='orderId'>
            {MyProfileData.constData.order_id}
            <span>{item.order_id}</span>
          </div>
          <button className='cancelOrder text-danger'>{MyProfileData.constData.cancelOrder}</button>
        </div>
        <div className='d-flex justify-content-between m-3'>
          <div className='col-4'>
            <img src={item.image} className='orderImage' alt={item.issue} />
          </div>
          <div className='col-8'>
            <div className='d-flex flex-row'>
              <i className='fa-solid fa-clock align-self-center mx-2' />
              <span className='fs-5'>{item.time}</span>
            </div>
            <div>
              <i className='fa-solid fa-laptop-code align-self-center mx-2' />
              <span className='fs-5'>{item.device}</span>
            </div>
            <div>
              <i className='fa-solid fa-gears align-self-center mx-2' />
              <span className='fs-5'>{item.issue}</span>
            </div>
          </div>
        </div>
        <p>{item.order_status}</p>
        <div className='d-flex flex-row justify-content-between'>
          <button className='callSupport'>{MyProfileData.constData.callSupport}</button>
          <button className='viewDetails'>{MyProfileData.constData.viewDetails}</button>
        </div>
      </section>
    </>
  )
}

export default OrderCard
