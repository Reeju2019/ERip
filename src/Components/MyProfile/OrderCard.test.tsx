import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'
import OrderCard from './OrderCard'


interface Data {
    order_id: string
    time: string
    device: string
    issue: string
    order_status: string
    image: string
}
const data: Data = {
    order_id: "",
    time: "",
    device: "",
    issue: "",
    order_status: "",
    image: "",
}

test('order card', () => {
  render(<OrderCard item={data}/>)
})
describe('order card rendering specification', () => {
  it('SnapShot testing', () => {
    const component = renderer.create(<OrderCard item={data}/>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('render without crashing order card', () => {
    const div = document.createElement('div')
    render(<OrderCard item={data}/>)
    ReactDOM.unmountComponentAtNode(div)
  })
})
