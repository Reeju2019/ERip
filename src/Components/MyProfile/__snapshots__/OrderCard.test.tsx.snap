// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`order card rendering specification SnapShot testing 1`] = `
<section
  className="bg-white p-3 orderCard"
>
  <div
    className="d-flex justify-content-between mt-3"
  >
    <div
      className="orderId"
    >
      Order ID: #
      <span />
    </div>
    <button
      className="cancelOrder text-danger"
    >
      Cancel Order
    </button>
  </div>
  <div
    className="d-flex justify-content-between m-3"
  >
    <div
      className="col-4"
    >
      <img
        alt=""
        className="orderImage"
        src=""
      />
    </div>
    <div
      className="col-8"
    >
      <div
        className="d-flex flex-row"
      >
        <i
          className="fa-solid fa-clock align-self-center mx-2"
        />
        <span
          className="fs-5"
        />
      </div>
      <div>
        <i
          className="fa-solid fa-laptop-code align-self-center mx-2"
        />
        <span
          className="fs-5"
        />
      </div>
      <div>
        <i
          className="fa-solid fa-gears align-self-center mx-2"
        />
        <span
          className="fs-5"
        />
      </div>
    </div>
  </div>
  <p />
  <div
    className="d-flex flex-row justify-content-between"
  >
    <button
      className="callSupport"
    >
      Call Support
    </button>
    <button
      className="viewDetails"
    >
      View Details
    </button>
  </div>
</section>
`;
